### PR TITLE
Add GCE workflow and instructions

### PR DESCRIPTION
**Pull request by rakyll**
GCE deployment currently focuses only on container-optimized VMs as noted in the instructions.

### CODE REVIEW

File: example-workflows/gce/.github/gce.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+name: Build and Deploy to Google Compute Engine
+
+on:
+  push:
+    branches:
+    - master
+
+env:
+  GCE_PROJECT: ${{ secrets.GCE_PROJECT }}
```
Comment (rakyll) [inline example-workflows/gce/.github/gce.yaml]: I'd be nice to standardize some of these variable names at some point. Each workflow is using something else and making it harder to reuse.

File: example-workflows/gce/.github/gce.yaml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    # Setup gcloud CLI
+    - uses: GoogleCloudPlatform/github-actions/setup-gcloud@master
+      with:
+        version: '270.0.0'
+        service_account_email: ${{ secrets.SA_EMAIL }}
+        service_account_key: ${{ secrets.GOOGLE_APPLICATION_CREDENTIALS }}
+
+    # Configure docker to use the gcloud command-line tool as a credential helper
+    - run: |
+        gcloud auth configure-docker
```
Comment (craigdbarber) [inline example-workflows/gce/.github/gce.yaml]: Docker is a pretty heavy weight setup in that it takes a long time to install on the runner. Suggest consider using `gcloud builds submit` as an alternative.

File: example-workflows/gce/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+For pushes to the _default_ branch, `master`, the workflow will:
+
+1. Verify the Google Cloud Platform credentials are correct.
+1. Build, tag, and push the image to Google Container Registry.
+1. Deploy the container image to a Google Compute Engine instance.
+    * Note that a GCE deployment requires an existing [container-optimized VM][create-vm].
+
+## Setup
+
+1. Create or reuse a Github repository for the example workflow:
```
Comment (craigdbarber) [inline example-workflows/gce/README.md]: Was there meant to be a link after "workflow:"?

File: example-workflows/gce/README.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+  1. [Create a repository](https://help.github.com/en/github/creating-cloning-and-archiving-repositories/creating-a-new-repository).
```
Comment (craigdbarber) [inline example-workflows/gce/README.md]: The indentation isn't consistent with the line item above.

File: example-workflows/gce/.github/gce.yaml (hunk 2 above)
Comment (rakyll) [reply example-workflows/gce/.github/gce.yaml]: I didn't want to enforce the user to enable Cloud Build to use this workflow. I assume advanced users can always switch to `gcloud builds submit` by modifying this workflow.

File: example-workflows/gce/README.md (hunk 3 above)
Comment (rakyll) [reply example-workflows/gce/README.md]: The following nested bulletin item list is what it's referring to.

**Comment by rakyll** [comment]
GitHub ate one of your comments but I also addressed it. There were nested lists in the doc but they were not rendered correctly due to bad formatting.

**Pull request merged**